### DISCUSSION TITLE
Upstash Redis for memory: how to tie SessionIDs and users?

### DISCUSSION THREAD

**Discussion by artemdzyuba** [FlowiseAI/Flowise#1063] [Q&A] [answered]
Hi! I was fascinated by the ease of using Upstash Redis for memory. It is supposed to remember the context even if the server restarts. 

So I left the SessionID field empty and restarted the server. Unfortunately, it starts a new session, I guess, because of the empty field. LangChain [recommends](https://js.langchain.com/docs/modules/memory/integrations/upstash_redis) using `new Date().toISOString()` code as a session ID. But in Flowise, not clear how to use expressions like that in fields. 

Another thing, how to tie different sessions for different users. Like when chatting via browsers or via API, Redis should remember the origin of the conversations. How to link them?

**Comment by HenryHengZJ** [comment, marked as answer]
do you mean from Flowise API or the chat embed?

If you are using API, you have to generate an ID to be used as `sessionId` from your front end apps. If you are using Chat Embed, Flowise will automatically use the chat message id as the sessionId. Currently when you refresh the page, the chat will lose all the conversation, but we have a PR that is going to persist the conversation

**Reply by artemdzyuba** [reply]
Yeah, I meant both. And I'll check your answer, but I see I can pass a session ID as an API call parameter.
